core/object/class_db.cpp

```diff
@@ -2155,4 +2163,45 @@ void ClassDB::cleanup() {
 	native_structs.clear();
 }
 
-//
+void ClassDB::tag_collect_pass(Object *p_object, uint32_t p_pass, bool p_containers) {
+	ClassInfo *check = classes.getptr(p_object->get_class_name());
+	while (check) {
+		for (uint32_t i = 0; i < check->resource_properties.size(); i++) {
+			Ref<Resource> res = p_object->get(check->resource_properties[i]);
+			if (res.is_valid()) {
+				res->tag_collect_pass(p_pass, p_containers);
+			}
+		}
+		if (p_containers) {
+			for (uint32_t i = 0; i < check->dict_properties.size(); i++) {
+				Dictionary dict = p_object->get(check->dict_properties[i]);
+				dict.tag_collect_pass(p_pass, p_containers);
+			}
+
+			for (uint32_t i = 0; i < check->array_properties.size(); i++) {
+				Array arr = p_object->get(check->array_properties[i]);
+				arr.tag_collect_pass(p_pass, p_containers);
+			}
+		}
+		check = check->inherits_ptr;
+	}
+}
+
+void ClassDB::unlink_resources(Object *p_object, bool p_containers) {
+	ClassInfo *check = classes.getptr(p_object->get_class_name());
+	while (check) {
+		for (uint32_t i = 0; i < check->resource_properties.size(); i++) {
+			p_object->set(check->resource_properties[i], Ref<Resource>());
+		}
+		if (p_containers) {
+			for (uint32_t i = 0; i < check->dict_properties.size(); i++) {
+				p_object->set(check->dict_properties[i], Dictionary());
+			}
+
+			for (uint32_t i = 0; i < check->array_properties.size(); i++) {
+				p_object->set(check->dict_properties[i], Array());
```

Comment:
```suggestion
		for (const StringName &prop : check->resource_properties) {
			p_object->set(prop, Ref<Resource>());
		}
		if (p_containers) {
			for (const StringName &prop : check->dict_properties) {
				p_object->set(prop, Dictionary());
			}

			for (const StringName &prop : check->array_properties) {
				p_object->set(prop, Array());
```
Also fixes the above bug